### PR TITLE
update modal.ts

### LINKED ISSUE #16722: AzureDataStudioPostgresExtension_ConnectionDetails_Keyboard:When we Tab till connection type dropdown and select postgreSQL and press enter the focus shifts to recent tab item .
https://user-images.githubusercontent.com/69922333/129091678-940793fe-12fb-40d9-9874-c09f11cbc564.mp4

Version: 1.31.1 (user setup)
Commit: 65fb22cc7c36db9c53af1ed2fdbdf48f66c682be
Date: 2021-07-28T21:03:21.018Z
VS Code: 1.56.0
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19043

**Repro Steps:**
Launch the application-Azure Data Studio and install extension PostgreSQL.
Tab till new connection button and press enter.
Tab till connection type dropdown
Press enter to see list and then select postgreSQL.
Check where the next focus is present after selecting postrgreSQL.

**Actual:**
When we Tab till connection type dropdown and select postgreSQL and press enter focus shifts to recent tab item which is not correct .

**Expected:**
When we Tab till connection type dropdown and select postgreSQL the focus should remain on the connection type dropdown only.

**User Impact:**
Keyboard users will face difficulty in accessing the screen.

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #16722 

when the connection type is changed, the event handler will eventually call the setter messagesElementVisible and at the moment the activeElement is still on the dropdown's popup as shown below and caused it to reset focus to first focusable element.

image

fix: this logic was originally introduced by me, we actually only need to make sure the focus is reset properly when the focus is currently in the messages area that will be removed.

### CODE REVIEW

File: src/sql/workbench/browser/modal/modal.ts [hunk 1]
```diff
@@ -598,10 +598,14 @@ export abstract class Modal extends Disposable implements IThemable {
 				DOM.prepend(this._modalContent!, this._messageElement!);
 			}
 		} else {
-			DOM.removeNode(this._messageElement!);
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/modal/modal.ts:601]: this method is deprecated, so replace it with the new method while i am here.

**Comment by alanrenmsft** [comment]
also tested with the Edge deployment page and pressed the close message button to close the message area.

**Review by Charles-Gagnon — APPROVED** [review]
